borrowd_groups/models.py

```diff
@@ -65,18 +101,43 @@ class BorrowdGroup(Group, GuardianGroupMixin):  # type: ignore[misc]
         help_text="The date and time at which the group was last updated.",
     )
 
+    # Override default manager to have custom `create()` method,
+    # which allows us to pass the trust level to the Membership
+    # model via the post_save signal.
+    # mypy error: Cannot override class variable (previously declared on base class "Group") with instance variable  [misc]
+    # ... but, this is a class variable, not an instance variable, right?
+    objects: BorrowdGroupManager = BorrowdGroupManager()  # type: ignore[misc]
+
     def get_absolute_url(self) -> str:
         return reverse("borrowd_groups:group-detail", args=[self.pk])
 
-    def add_user(self, user: BorrowdUser, trust_level: TrustLevel) -> "Membership":
+    def add_user(
+        self, user: BorrowdUser, trust_level: TrustLevel, is_moderator: bool = False
+    ) -> "Membership":
         """
         Add a user to the group.
         """
-        return Membership.objects.create(
-            user=user,
-            group=self,
-            trust_level=trust_level,
-        )
+        membership: Membership
+        # TODO: Should this simply be an .update_or_create() call?
+        # I'm opting _not_ at this point, because we should never
+        # logically be in a position where we're _trying_ to add
+        # a user to a Group they're already in: so if that happens,
+        # we should want to diagnose the situation that led us there.
+        # But, open to other opinions.
```

Comment:
In addition to the reason that you specified, I've seen `update_or_create` get buried in logic and cause massive performance issues when used at scale. I don't think we'd have that sort of issue here, but it seems best to avoid it in general and just be more explicit.